rudder-core/src/main/scala/com/normation/cfclerk/services/impl/GitTechniqueReader.scala

```diff
@@ -748,14 +745,29 @@ class GitTechniqueReader(
    * Root is "/", so that a package "P1" is denoted
    * /P1, a package P2 in sub category cat1 is denoted
    * /cat1/P2, etc.
+   * As we may have files&templates outside technique, we
+   * need to keep track of there relative id
    */
-  private[this] def getTechniquePath(techniqueInfos:TechniquesInfo) : Set[TechniquePath] = {
-   var set = scala.collection.mutable.Set[TechniquePath]()
-   techniqueInfos.rootCategory.packageIds.foreach { p => set += TechniquePath( "/" + p.toString) }
-   techniqueInfos.subCategories.foreach { case (id,cat) =>
-     val path = id.toString
-     cat.packageIds.foreach { p => set += TechniquePath(path + "/" + p.toString) }
+  private[this] def getTechniquePath(techniqueInfos:TechniquesInfo) : Set[(TechniquePath, TechniqueId)] = {
+    var set = scala.collection.mutable.Set[(TechniquePath, TechniqueId)]()
+    techniqueInfos.rootCategory.packageIds.foreach { id =>
+      set += ((TechniquePath( "/" + id.toString), id))
+    }
+    techniqueInfos.subCategories.foreach { case (id,cat) =>
+      val path = id.toString
+      cat.packageIds.foreach { t => set += ((TechniquePath(path + "/" + t.toString), t)) }
+    }
+    //also add template "by path"
+    val techniques = techniqueInfos.techniques.flatMap { case(_, set) => set.map { case(_, t) => t } }
+    techniques.foreach { t =>
+      val byPath = t.templates.collect { case TechniqueTemplate(id@TechniqueResourceIdByPath(_,_),_,_) => id } ++
+                   t.files.collect { case TechniqueFile(id@TechniqueResourceIdByPath(_,_),_,_) => id }
+      byPath.foreach { x =>
```

Comment:
Not at ease with the x variable here ... but i can understand it's hard to name it with a understandable name :)
resource ? 
techniqueResource? 
x ? :) 

Maybe it miss a type or a better name for byPath variable so we understand what we handle here